apps/src/javalab/JavalabView.jsx

```diff
@@ -84,10 +84,12 @@ class JavalabView extends React.Component {
     this.props.appendOutputLog('Compiled!');
   };
 
+  // Sends redux call to update dark mode, which handles user preferences
   renderSettings = () => {
     const {isDarkMode} = this.props;
+    this.setIsDarkMode = this.props.setIsDarkMode.bind(this, !isDarkMode);
     return [
-      <a onClick={this.props.toggleDarkMode} key="theme-setting">
+      <a onClick={this.setIsDarkMode} key="theme-setting">
```

Comment:
nit: I think we can call `this.props.setIsDarkMode` directly here like `() => this.props.setIsDarkMode(!isDarkMode)`

Reply:
Ah yes- updating now. Thanks!